Source/WebCore/Modules/compression/CompressionStreamEncoder.cpp

```diff
@@ -164,4 +176,82 @@ ExceptionOr<Ref<JSC::ArrayBuffer>> CompressionStreamEncoder::compress(std::span<
 
     return compressedData.releaseNonNull();
 }
+
+
+#if PLATFORM(COCOA)
+ExceptionOr<bool> CompressionStreamEncoder::initializeAppleCompressionFramework() 
+{
+    m_initialized = true;
+    m_usingAppleCompressionFramework = true;
+
+    auto result = compression_stream_init(&m_stream, COMPRESSION_STREAM_ENCODE, COMPRESSION_BROTLI);
+    if (result != COMPRESSION_STATUS_OK)
+        return Exception { ExceptionCode::TypeError, "Initialization Failed."_s };
+
+    return true;
+}
+
+ExceptionOr<Ref<JSC::ArrayBuffer>> CompressionStreamEncoder::compressAppleCompressionFramework(std::span<const uint8_t> input)
+{
+    m_didFinish = true;
+
+    size_t allocateSize = std::max(input.size(), startingAllocationSize);
+    auto storage = SharedBufferBuilder();
+
+    compression_status result;    
+    bool shouldDecompress = true;
+
+    if (!m_initialized) {
+        auto initializeResult = initializeAppleCompressionFramework();
+        if (initializeResult.hasException())
+            return initializeResult.releaseException();
+    }
+
+    m_stream.src_ptr = input.data();
+    m_stream.src_size = input.size();
+    
+    while (shouldDecompress) {
+        Vector<uint8_t> output;
+        if (!output.tryReserveInitialCapacity(allocateSize)) {
+            allocateSize /= 4;
```

Comment:
4 seems like a very large factor to shrink by if allocation fails.  That seems like it'll make it very likely to be too small.